File: src/main/java/priv/bajdcc/util/Position.java
package priv.bajdcc.util;

/**
 * 位置
 *
 * @author bajdcc
 */
public class Position implements Cloneable {
	/**
	 * 列号
	 */
	public int iColumn = 0;

	/**
	 * 行号
	 */
	public int iLine = 0;

	public Position() {
	}

	public Position(Position obj) {
		iColumn = obj.iColumn;
		iLine = obj.iLine;
	}

	public Position(int col, int line) {
		iColumn = col;
		iLine = line;
	}

	@Override
	public String toString() {
		return iLine + "," + iColumn;
	}

	@Override
	public Object clone() throws CloneNotSupportedException {
		return super.clone();
	}
}

File: src/main/java/com/bajdcc/util/Position.kt
package com.bajdcc.util

/**
 * 位置
 *
 * @author bajdcc
 */
class Position : Cloneable {
    /**
     * 列号
     */
    var column = 0

    /**
     * 行号
     */
    var line = 0

    constructor()

    constructor(obj: Position) {
        column = obj.column
        line = obj.line
    }

    constructor(column: Int, line: Int) {
        this.column = column
        this.line = line
    }

    override fun toString(): String {
        return "$line,$column"
    }

    public override fun clone(): Any {
        return super.clone()
    }

    fun different(obj: Position): Boolean {
        return this.line != obj.line || this.column != obj.column
    }
}
